### DISCUSSION TITLE
如何理解Downsampling（降采样）?

### DISCUSSION THREAD

**Discussion by chengjianyun** [apache/iotdb#3976] [Q&A] [answered]
IoTDB官网介绍`Downsampling` 如下:

> Downsampling：
>
> Downsampling（降采样）用于改变时间序列的粒度，例如：从 10Hz 到 1Hz，或者每天 1 个点。
> 
> 和其他数据库不同的是，IoTDB 能够实时降采样数据，而其它时间序列数据库在磁盘上序列化降采样数据。
> 
> 也就是说：
> 
> IoTDB 支持在任意时间对数据进行即席（ad-hoc）降采样。例如：一条 SQL 返回从 2020-04-27 08:00:00 开始的每 5 分钟采样 1 个点的降采样数据，另一条 SQL 返回从 2020-04-27 08:00:01 开始的每 5 分 10 秒采样 1 个点的降采样数据。
> 
> （InfluxDB 也支持即席降采样，但是性能似乎并不好。）
> 
> IoTDB 的降采样不占用磁盘。

有几个点不是很清楚：

1. 降采样指的是在数据存储时就保存降频后的数据吗？如果是这样，对应的数据插入接口应该使用哪个？
2. 如何理解即席降采样？从给出的例子中很难理解。
3. 降采样不占用磁盘指的是什么？

**Comment by qiaojialin** [comment, marked as answer]
1. 写入数据还是全量存储的。
2. 即席降采样其实是 group by 查询搭配 last_value 聚合函数时的语义。 group by 是分段聚合查询，查询时指定分段的起始点，分段的间隔和滑动步长，返回每一段时间的聚合结果，结果中time以段的开头或结尾表示，value是这一段数据的某种聚合结果，比如 avg 是每段数据的平均值，last_value 就是每段的最后一个值（这个就是降采样查询了）。
3. 查询时降采样不需要持久化存储，是跟物化视图的区别。
